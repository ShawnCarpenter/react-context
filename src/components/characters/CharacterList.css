.CharacterList {
  display: grid;
  grid-template:repeat(4, 1fr) / repeat(5, 1fr);
  align-items: baseline;
  img {
    max-width: 100px;
    border-radius:.3rem;
  }

  li {
    list-style: none;
  }
}
