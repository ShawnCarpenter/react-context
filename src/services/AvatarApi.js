export const getCharacters = async(page = 1) => {
  // eslint-disable-next-line max-len
  const res = await fetch(`https://last-airbender-api.herokuapp.com/api/v1/characters/?perPage=20&page=6`);
  
  if(!res.ok)  throw 'Uh-oh, something done broked!';

  const json = await res.json();

  return json.map(character => ({
    _id:character._id,
    name:character.name,
    photoUrl:character.photoUrl
  }));
};
