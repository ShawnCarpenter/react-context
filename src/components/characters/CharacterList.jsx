import React from 'react';
import { useCharacters } from '../../hooks/characters';
import { useTheme } from '../../hooks/theme';
import CharacterItem from './CharacterItem';
import styles from './CharacterList.css';

export const CharacterList = () => {
  const { loading, characters } = useCharacters();
  const theme = useTheme();
  
  if(loading) return <h1>Loading...</h1>;
  const characterElements = characters.map(character => (
    <li key={character._id}>
      <CharacterItem {...character} />
    </li>
  ));
  return (
    <ul data-testid="characters" className={styles.CharacterList}
      style={theme}>
      {characterElements}
    </ul>
  );
};
