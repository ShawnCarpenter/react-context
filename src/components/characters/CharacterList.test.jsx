import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import { CharacterList } from './CharacterList';
import { getCharacters } from '../../services/AvatarApi';

jest.mock('../../hooks/theme.js');
jest.mock('../../services/AvatarApi.js');


describe('CharacterList component', () => {
  it('renders CharacterList', async() => {
    getCharacters.mockResolvedValue([{
      _id:'id',
      name:'Dude',
      photoUrl:'http://www.img.com/pic.jpg'
    }]);

    const { asFragment } = render(
      <CharacterList />
    );
    screen.getByText('Loading...');
    const characterList = await screen.findByTestId('characters');
    expect(asFragment()).toMatchSnapshot();
    return waitFor(() => {
      expect(characterList).not.toBeEmptyDOMElement();
      expect(characterList).toHaveTextContent('Dude');

    });
  });
});
