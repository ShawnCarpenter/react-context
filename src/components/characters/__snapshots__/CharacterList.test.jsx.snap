// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterList component renders CharacterList 1`] = `
<DocumentFragment>
  <ul
    class="CharacterList"
    data-testid="characters"
  >
    <li>
      <figure>
        <img
          alt="Dude"
          src="http://www.img.com/pic.jpg"
        />
        <figcaption>
          Dude
        </figcaption>
      </figure>
    </li>
  </ul>
</DocumentFragment>
`;
